src/App.tsx

```diff
@@ -37,14 +37,35 @@ function App(): React.ReactElement {
   const [dataProvider, setDataProvider] = useState<DataProvider | undefined>(
     undefined
   )
-
+  const [loggingPref, setLoggingPref] = useState<null | string>(null)
```

Comment:
hello @TahaKhanAbdalli - while the value is stored as string in localStorage, I'd like to hide that from the rest of the app, please. So, I'd like the state to be boolean (default value false), and when we retrieve it from `localStorage` immediately convert to a boolean.

The big picture here is that the fact it's stored as a string in the browser is a detail that the rest of the app doesn't need to know about :-)